mfr/extensions/zip/render.py

```diff
@@ -0,0 +1,31 @@
+import os
+import zipfile
+import markupsafe
+
+from mako.lookup import TemplateLookup
+
+from mfr.core import extension
+
+
+class ZipRenderer(extension.BaseRenderer):
+
+    TEMPLATE = TemplateLookup(
+        directories=[
+            os.path.join(os.path.dirname(__file__), 'templates')
+        ]).get_template('viewer.mako')
+
+    def render(self):
+        zip_file = zipfile.ZipFile(self.file_path, 'r')
+
+        return self.TEMPLATE.render(zipped_filenames=self.format_zip(zip_file))
+
+    def format_zip(self, zip_file):
+        return '<br>'.join([markupsafe.escape(x) for x in zip_file.namelist()])
```

Comment:
Both `namelist()` and `infolist()` contain duplicated entries. For example, `mfr-test.txt` has three entries where `__MACOSX/` and `__MACOSX/.__mfr-test.txt` are not what we want.  In addition, unlike `file`, `zip` and `dir` do not have the prefix. Is there a general rule that we know?

What's in my zip:

```
mfr-test/
mfr-test.zip
mfr-test.docx
mfr-test.pdf
mfr-test.pptx
mfr-test.txt
_mfr-test.txt
```
What shows up in OSF:

```
_mfr-test.txt
__MACOSX/
__MACOSX/.__mfr-test.txt
mfr-test/
mfr-test.docx
__MACOSX/._mfr-test.docx
mfr-test.pdf
__MACOSX/._mfr-test.pdf
mfr-test.pptx
__MACOSX/._mfr-test.pptx
mfr-test.txt
__MACOSX/._mfr-test.txt
mfr-test.zip
```

Reply:
Another concern is that the prefix seems OS dependent. If we want to identify it, should we consider taking other OS into account?